pkg/apis/internal.admin.acorn.io/v1/types.go

```diff
@@ -0,0 +1,62 @@
+// +k8s:deepcopy-gen=package
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterWorkloadClassInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec WorkloadClassSpec `json:"spec,omitempty"`
+}
+
+func (in *ClusterWorkloadClassInstance) NamespaceScoped() bool {
+	return false
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterWorkloadClassInstanceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ClusterWorkloadClassInstance `json:"items"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ProjectWorkloadClassInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec WorkloadClassSpec `json:"spec,omitempty"`
```

Comment:
I was using the spec as way to generically get WorkloadClass information. I realize that Spec was not needed for this nor conventional now and replaced it with a new type - `WorkloadClassInstance`. It contains all of the values that will be used for `ProjectWorkloadClass` and `ClusterWorkloadClass` so that you can easily cast them into it. Thoughts?

Reply:
Updated WorkloadClassInstance to be replaced by just using ProjectWorkloadClassInstance instead.